### ISSUE TITLE
dependency on numpy is missing from setup.py

### ISSUE THREAD

**Issue by simonvanderveldt** [dask/fastparquet#389]
Numpy is missing in the list of dependencies in `setup.py` causing a installation to fail when numpy isn't installed before.
Even if installation were possible running code which uses fastparquet would of course fail when numpy wouldn't be installed as well :)
Is there a reason the dependency on numpy missing?

**Comment by martindurant** [comment]
Numpy *should* be a dependency.

**Comment by josh-theorem** [comment]
Also running into this. It looks like this hack:

https://github.com/dask/fastparquet/blob/master/setup.py#L12-L23

isn't working as expected. I'm still getting a `No module named numpy` import error.

**Comment by martindurant** [comment]
@josh-theorem , happy to consider any fix you have. I am not a master of setup things, unfortunately.

**Comment by javabrett** [comment]
This problem can be quickly reproduced with a Docker build:
```
tee Dockerfile <<EOF
FROM python:3

COPY . /fastparquet
RUN pip install /fastparquet
EOF
```

<details>
<summary> docker build --no-cache -t fastparquet .</summary>

```
Sending build context to Docker daemon  152.6MB
Step 1/3 : FROM python:3
 ---> 338b34a7555c
Step 2/3 : COPY . /fastparquet
 ---> 990cd6f1b98a
Step 3/3 : RUN pip install /fastparquet
 ---> Running in 626fbd5b6882
Processing /fastparquet
Collecting pandas>=0.19 (from fastparquet==0.2.1)
  Downloading https://files.pythonhosted.org/packages/c0/f7/7fd96f43d84a22d068fc999f2ae27f677be4c9501441f5b7870b75842fa4/pandas-0.24.1-cp37-cp37m-manylinux1_x86_64.whl (10.1MB)
Collecting numba>=0.28 (from fastparquet==0.2.1)
  Downloading https://files.pythonhosted.org/packages/15/ea/6c14dc1039b662bca8ec62a95d17b74fd3b1e53fabaff918d691e0efbf65/numba-0.42.1-cp37-cp37m-manylinux1_x86_64.whl (3.2MB)
Collecting numpy>=1.11 (from fastparquet==0.2.1)
  Downloading https://files.pythonhosted.org/packages/61/57/07c49e1a6d2706fb7336b3fb11dd285c1e96535c80833d7524f002f57086/numpy-1.16.1-cp37-cp37m-manylinux1_x86_64.whl (17.3MB)
----- 8-< ----- 8-< ----- 8-< ----- 8-< ----- 8-< -----
Building wheels for collected packages: fastparquet, thrift
  Building wheel for fastparquet (setup.py): started
  Building wheel for fastparquet (setup.py): finished with status 'error'
  Complete output from command /usr/local/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-43o6j8fq/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-xlfqm3_f --python-tag cp37:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/fastparquet
  copying fastparquet/core.py -> build/lib.linux-x86_64-3.7/fastparquet
  copying fastparquet/writer.py -> build/lib.linux-x86_64-3.7/fastparquet
  copying fastparquet/dataframe.py -> build/lib.linux-x86_64-3.7/fastparquet
  copying fastparquet/converted_types.py -> build/lib.linux-x86_64-3.7/fastparquet
  copying fastparquet/compression.py -> build/lib.linux-x86_64-3.7/fastparquet
  copying fastparquet/util.py -> build/lib.linux-x86_64-3.7/fastparquet
  copying fastparquet/encoding.py -> build/lib.linux-x86_64-3.7/fastparquet
  copying fastparquet/thrift_structures.py -> build/lib.linux-x86_64-3.7/fastparquet
  copying fastparquet/__init__.py -> build/lib.linux-x86_64-3.7/fastparquet
  copying fastparquet/api.py -> build/lib.linux-x86_64-3.7/fastparquet
  copying fastparquet/schema.py -> build/lib.linux-x86_64-3.7/fastparquet
  running egg_info
  creating fastparquet.egg-info
  writing fastparquet.egg-info/PKG-INFO
  writing dependency_links to fastparquet.egg-info/dependency_links.txt
  writing requirements to fastparquet.egg-info/requires.txt
  writing top-level names to fastparquet.egg-info/top_level.txt
  writing manifest file 'fastparquet.egg-info/SOURCES.txt'
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-req-build-43o6j8fq/setup.py", line 102, in <module>
      **extra
    File "/usr/local/lib/python3.7/site-packages/setuptools/__init__.py", line 145, in setup
      return distutils.core.setup(**attrs)
    File "/usr/local/lib/python3.7/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/local/lib/python3.7/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/usr/local/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/usr/local/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 188, in run
      self.run_command('build')
    File "/usr/local/lib/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/local/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/usr/local/lib/python3.7/distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/usr/local/lib/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/local/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/usr/local/lib/python3.7/site-packages/setuptools/command/build_py.py", line 53, in run
      self.build_package_data()
    File "/usr/local/lib/python3.7/site-packages/setuptools/command/build_py.py", line 118, in build_package_data
      for package, src_dir, build_dir, filenames in self.data_files:
    File "/usr/local/lib/python3.7/site-packages/setuptools/command/build_py.py", line 66, in __getattr__
      self.data_files = self._get_data_files()
    File "/usr/local/lib/python3.7/site-packages/setuptools/command/build_py.py", line 82, in _get_data_files
      self.analyze_manifest()
    File "/usr/local/lib/python3.7/site-packages/setuptools/command/build_py.py", line 138, in analyze_manifest
      self.run_command('egg_info')
    File "/usr/local/lib/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/local/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/usr/local/lib/python3.7/site-packages/setuptools/command/egg_info.py", line 296, in run
      self.find_sources()
    File "/usr/local/lib/python3.7/site-packages/setuptools/command/egg_info.py", line 303, in find_sources
      mm.run()
    File "/usr/local/lib/python3.7/site-packages/setuptools/command/egg_info.py", line 534, in run
      self.add_defaults()
    File "/usr/local/lib/python3.7/site-packages/setuptools/command/egg_info.py", line 570, in add_defaults
      sdist.add_defaults(self)
    File "/usr/local/lib/python3.7/distutils/command/sdist.py", line 228, in add_defaults
      self._add_defaults_ext()
    File "/usr/local/lib/python3.7/distutils/command/sdist.py", line 311, in _add_defaults_ext
      build_ext = self.get_finalized_command('build_ext')
    File "/usr/local/lib/python3.7/distutils/cmd.py", line 299, in get_finalized_command
      cmd_obj.ensure_finalized()
    File "/usr/local/lib/python3.7/distutils/cmd.py", line 107, in ensure_finalized
      self.finalize_options()
    File "/tmp/pip-req-build-43o6j8fq/setup.py", line 22, in finalize_options
      import numpy
  ModuleNotFoundError: No module named 'numpy'
  
  ----------------------------------------
  Failed building wheel for fastparquet
  Running setup.py clean for fastparquet
  Building wheel for thrift (setup.py): started
  Building wheel for thrift (setup.py): finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/be/36/81/0f93ba89a1cb7887c91937948519840a72c0ffdd57cac0ae8f
Successfully built thrift
Failed to build fastparquet
Installing collected packages: pytz, six, python-dateutil, numpy, pandas, llvmlite, numba, thrift, pytest-runner, fastparquet
  Running setup.py install for fastparquet: started
    Running setup.py install for fastparquet: finished with status 'done'
Successfully installed fastparquet-0.2.1 llvmlite-0.27.1 numba-0.42.1 numpy-1.16.1 pandas-0.24.1 pytest-runner-4.4 python-dateutil-2.8.0 pytz-2018.9 six-1.12.0 thrift-0.11.0
You are using pip version 19.0.1, however version 19.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Removing intermediate container 626fbd5b6882
 ---> c56917d5eac2
Successfully built c56917d5eac2
Successfully tagged fastparquet:latest
```
</details>

Key output:
```
File "/tmp/pip-req-build-43o6j8fq/setup.py", line 22, in finalize_options
      import numpy
  ModuleNotFoundError: No module named 'numpy'
```

The original `numpy` setup requires workaround was applied in 71d825029809505acb4111c98d6604dff52055ba and fixed in `0.1.4`.  However, commit 14a6af65b96559d8834948bf32d375d864476b23 removes the `setup_requires` block for `numpy` in `0.1.6` and the problem returns.

See also #345.  It claims `setup_requires` needs to be removed for corporate CI/CD environments, since `easy_install` goes straight to the standard pypi index and can't be controlled.  That assertion needs verifying.

Reverting the commit, yielding the following diff:
<details>
<summary> git diff</summary>

```

diff --git a/setup.py b/setup.py
index 4caa7e7..e73d369 100644
--- a/setup.py
+++ b/setup.py
@@ -49,9 +49,6 @@ else:
 
 install_requires = open('requirements.txt').read().strip().split('\n')
 
-# add pytest-runner from setup_requires
-install_requires.append('pytest-runner')
-
 setup(
     name='fastparquet',
     version='0.2.1',
@@ -76,11 +73,10 @@ setup(
     packages=['fastparquet'],
     cmdclass={'build_ext': build_ext},
     install_requires=install_requires,
-    # remove wierd setup_requires to get rid of easy_install
-    # setup_requires=[
-    #    'pytest-runner',
-    #    [p for p in install_requires if p.startswith('numpy')][0]
-    #],
+    setup_requires=[
+        'pytest-runner',
+        [p for p in install_requires if p.startswith('numpy')][0]
+    ],
     extras_require={
         'brotli': ['brotli'],
         'lz4': ['lz4 >= 0.19.1'],
```
</details>

... and the Docker test-build succeeds with the `fastparquet` wheel build, since `numpy` is forced to install first.
